common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
 {
+  // Used only for getting metadata storage connector password, present here for backwards compatibility
+  @Deprecated
   public String getPassword();
+
+  /**
+   * Implementations should provide a way to get password for the input key.
+   * For example {@link DefaultPasswordProvider} looks for a property named
+   * {@param key} prefixed with {@link DefaultPasswordProvider#PWD_PROPERTY_PREFIX}
+   * in the runtime properties and return the value.
+   */
+  public String getPassword(String key);
```

Comment:
What @jon-wei said makes sense to me… I think the idea with this interface is that the `key` is a well-known config name like `metadataStorePassword` or `tlsKeyStorePassword` and it's up to the PasswordProvider to handle that however it wants to.